### PR TITLE
Transition to use CRMI definitions for Measure and Library

### PR DESCRIPTION
**Pull request by elsaperelli**
# Summary
This PR updates the typing of Libraries and Measures by using (when possible) [CRMIShareableMeasure](https://hl7.org/fhir/uv/crmi/StructureDefinition-crmi-shareablemeasure.html) and [CRMIShareableLibrary](https://hl7.org/fhir/uv/crmi/StructureDefinition-crmi-shareablelibrary.html) type definitions. 

## New behavior
The CRMIShareableMeasure and CRMIShareableLibrary definitions only vary slightly from that of fhir4.Measure and fhir4.Library- they are both extended from them but also require `version`, `url`, `title`, `status` and `description`. 

## Code changes
- A lot of changing `fhir4.FhirResource` to `fhir4.Measure` and `fhir4.Library` to `CRMIShareableMeasure` and `CRMIShareableLibrary`. 
- A lot of making sure test fixtures have additional attributes (`url`, `version`, `title`, `description`)
- Deleting some no longer relevant unit tests

# Testing guidance
- `npm run check:all`
- `npm run build:all`
- `npm run start:all`
- Make sure the functionality is the same.
- Go through the code to make sure I didn't miss any places where the new types can be used. Note: I purposely did not change anything related to uploading bundles because I did not want to mess with the existing error handling there. Open to alternative ways to do this though.
- Make sure that all unit tests are still relevant.

### CODE REVIEW

**Review by hossenlopp — APPROVED** [review]
Looks good to me.

**Review by lmd59 — COMMENTED** [review]
Looks good! A couple of small recommendations in line as well as recommend replace `fhir4.Library` with `CRMIShareableLibrary` in app/src/server/trpc/routers/service.ts and app/src/pages/[resourceType]/[id].tsx

Also, I don't know if we necessarily need to, but thought I'd double check if this warrants an update to the service README. There's a lot of mention of `Measure` and `Library` related support, so do you think it's worth adding a bit about compliance with the CRMI versions of those resources?

File: app/src/util/types/fhir.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
-export interface CQFMMeasure extends fhir4.Measure {
+export interface CRMIShareableMeasure extends fhir4.Measure {
   id: string;
   version: string;
 }
 
-export interface CQFMLibrary extends fhir4.Library {
+export interface CRMIShareableLibrary extends fhir4.Library {
   id: string;
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply app/src/util/types/fhir.ts:7]: Good catch!

File: service/src/util/inputUtils.ts [hunk 2]
```diff
@@ -108,10 +109,7 @@ export function checkAuthoring() {
   }
 }
 
-export function checkFieldsForUpdate(
-  resource: fhir4.Measure | fhir4.Library,
-  oldResource: fhir4.Measure | fhir4.Library
-) {
+export function checkFieldsForUpdate(resource: FhirArtifact, oldResource: FhirArtifact) {
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply service/src/util/inputUtils.ts:112]: So I was sure about this one. I didn't want to mess with the error handling in this function that says specifically what fields are missing. I didn't change a lot of the uploading resource functionality for this reason. I also think I was just having a hard time thinking of another way to do this so definitely looking for suggestions!

[the remark this answers is not in the document]
Comment (elsaperelli) [reply service/src/util/inputUtils.ts:112]: Reverting `checkFieldForUpdate` to previous typing!

**Review by lmd59 — APPROVED** [review]
Nice!

**Review by hossenlopp — APPROVED** [review]

**Pull request merged**